processing/src/main/java/io/druid/segment/DimensionSelector.java

```diff
@@ -80,10 +90,19 @@
   public String lookupName(int id);
 
   /**
-   * The ID is the int id value of the field.
+   * Returns true if it is possible to {@link #lookupName(int)} by ids from 0 to {@link #getValueCardinality()}
+   * before the rows with those ids are returned.
    *
-   * @param name field name to look up the id for
-   * @return the id for the given field name
+   * <p>Returns false if {@link #lookupName(int)} could be called only with ids, returned from the most recent call of
+   * {@link #getRow()} on this DimensionSelector, but not earlier and not later. If {@link #lookupName(int)} is called
+   * with an id, which is not present in the most recent {@link IndexedInts}, returned from {@link #getRow()}, result is
+   * undefined: exception could be thrown, or null returned, or some other random value.
+   */
+  boolean nameLookupPossibleInAdvance();
```

Comment:
I think such change is a subject for a separate PR, which adds support of topN processing with DimensionSelectors without cardinality, and probably also removes the cache from `SingleScanTimeDimSelector`